src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/ChecklistServiceImpl.java

```diff
@@ -59,10 +66,202 @@ public ChecklistServiceImpl(ApplicationData applicationData, Tools tools, InforW
+				responsibilityToSignature.put(responsibilityCode, signaturesWithResp);
+				filters.add(new GridRequestFilter(
+						"responsibility", responsibilityCode, "=", GridRequestFilter.JOINER.OR, false, false));
+			}
+		}
+		if(filters.size() == 0) return;
+		gridRequest.setGridRequestFilters(filters);
+		GridRequestResult request = gridsService.executeQuery(context, gridRequest);
+		GridRequestRow[] rows = request.getRows();
+		GridRequestCell emptyCell = new GridRequestCell("", "");
+		BiFunction<GridRequestRow, String, String> getCellContent = (row, tag)->
+					Arrays.stream(row.getCell())
+							.filter(cell -> cell.getTag().equals(tag)).findFirst().orElse(emptyCell).getContent();
+
+		Arrays.stream(rows).forEach(row ->
+				responsibilityToSignature.get(getCellContent.apply(row, "responsibility")).stream()
+						.forEach(signature -> signature.setResponsibilityDescription(getCellContent.apply(row, "description"))));
+	}
+
+	private WorkOrderActivityChecklistSignatureResult[] filterSignatures(WorkOrderActivityCheckListDefaultResult workOrderActivityCheckList) throws Exception {
+		DateAdapter dateAdapter = new DateAdapter();
+		String reviewerQualification = workOrderActivityCheckList.getReviewerQualification();
+		String performer1Qualification = workOrderActivityCheckList.getPerformer1Qualification();
+		String performer2Qualification = workOrderActivityCheckList.getPerformer2Qualification();
+		List<String> qualifications = new ArrayList<>();
+		for(USERDEFINEDCODEID_Type userData : workOrderActivityCheckList.getUserResponsibilities())
+			qualifications.add(userData.getUSERDEFINEDCODE());
+
+		boolean noRequiredQualifications = reviewerQualification == null &&
+				                           performer1Qualification == null &&
+				                           performer2Qualification == null;
+
+		List<WorkOrderActivityChecklistSignatureResult> signatures = new LinkedList<>();
+
+		boolean isPerformer1 = noRequiredQualifications || performer1Qualification == null
+							   || qualifications.contains(performer1Qualification);
+
+		boolean isPerformer2 = noRequiredQualifications || performer2Qualification == null && isPerformer1
+				               || qualifications.contains(performer2Qualification);
+
+		boolean isReviewer = noRequiredQualifications || reviewerQualification == null && (isPerformer1 || isPerformer2)
+							 || qualifications.contains(reviewerQualification);
+
+		if(isPerformer1 || isReviewer){
+			WorkOrderActivityChecklistSignatureResult perf1 = new WorkOrderActivityChecklistSignatureResult();
+			perf1.setType("PB01");
+			perf1.setSigner(workOrderActivityCheckList.getPerformer1Name());
+			perf1.setViewAsPerformer(isPerformer1);
+			perf1.setViewAsReviewer(isReviewer);
+			perf1.setTime(workOrderActivityCheckList.getTimePerf1() != null ? dateAdapter.marshal(workOrderActivityCheckList.getTimePerf1()) : null);
+			perf1.setResponsibilityCode(performer1Qualification);
+			signatures.add(perf1);
+		}
```

Comment:
I feel this can be written as a function to make the code clearer. To be discussed.